### PR TITLE
clean up event handlers and remove handlers on clearEntities

### LINKED ISSUE #2: fix memory leak
https://github.com/peopleconnectus/ember-bing-maps/blob/master/app/components/bing-map.js#L83

There must be a way to remove handlers.

### PR DESCRIPTION
**Pull request by psheraton**
This fixes https://github.com/peopleconnectus/ember-bing-maps/issues/2

### CODE REVIEW

File: app/components/bing-map.js [hunk 1]
```diff
@@ -50,47 +59,63 @@ export default Component.extend({
...
-        Microsoft.Maps.Events.addHandler(pin, 'mousedown', (e) => onPinMouseDown(e, infobox, map));
-        Microsoft.Maps.Events.addHandler(pin, 'mouseout', (e) => onPinMouseOut(e, infobox, map));
-        Microsoft.Maps.Events.addHandler(pin, 'mouseover', (e) => onPinMouseOver(e, infobox, map));
-        Microsoft.Maps.Events.addHandler(pin, 'mouseup', (e) => onPinMouseUp(e, infobox, map));
+        this.addPinEvents(pin);
       });
     }
   },
 
   removeMap: function() {
```
Comment (kellyselden) [inline app/components/bing-map.js:117]: who calls removeMap?

Comment (psheraton) [reply app/components/bing-map.js]:
```willDestroyElement() {
    this.removeMap();
  },```

Comment (psheraton) [reply app/components/bing-map.js:117]: willDestroyElement() {
    this.removeMap();
  },

Comment (kellyselden) [reply app/components/bing-map.js:117]: perfect. I didn't see that in the diff.

**Pull request merged**